cmd/api/src/api/v2/auth/sso.go

```diff
@@ -0,0 +1,143 @@
+	for _, column := range sortByColumns {
+		var descending bool
+		if strings.HasPrefix(column, "-") {
+			descending = true
+			column = column[1:]
+		}
+
+		if !model.SSOProviderSortableFields(column) {
+			api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsNotSortable, request), response)
+			return
+		}
+
+		if descending {
+			order = append(order, column+" desc")
+		} else {
+			order = append(order, column)
+		}
+	}
+
+	// Set default order by created_at if no sorting is specified
+	if len(order) == 0 {
+		order = append(order, "created_at")
+	}
+
+	if queryFilters, err = queryFilterParser.ParseQueryParameterFilters(request); err != nil {
+		api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsBadQueryParameterFilters, request), response)
+		return
+	} else {
+		// Validate filters
+		for name, filters := range queryFilters {
+			if validPredicates, err := model.SSOProviderValidFilterPredicates(name); err != nil {
+				api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response)
+				return
+			} else {
+				for i, filter := range filters {
+					if !utils.Contains(validPredicates, string(filter.Operator)) {
+						api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest, api.ErrorResponseDetailsFilterPredicateNotSupported, request), response)
+						return
+					}
+					queryFilters[name][i].IsStringData = model.SSOProviderIsStringField(filter.Name)
+				}
+			}
+		}
+
+		if sqlFilter, err = queryFilters.BuildSQLFilter(); err != nil {
+			api.WriteErrorResponse(ctx, api.BuildErrorResponse(http.StatusBadRequest, "error building SQL for filter", request), response)
+			return
+		} else if ssoProviders, err = s.db.GetAllSSOProviders(ctx, strings.Join(order, ", "), sqlFilter); err != nil {
+			api.HandleDatabaseError(request, response, err)
+			return
+		} else {
+			for _, ssoProvider := range ssoProviders {
+				var details interface{}
+
+				switch ssoProvider.Type {
+				case model.SSOProviderTypeOIDC:
+					var oidcProvider model.OIDCProvider
+					if oidcProvider, err = s.db.GetOIDCProviderBySSOProviderID(ctx, int(ssoProvider.ID)); err != nil {
+						api.HandleDatabaseError(request, response, err)
+						return
+					} else {
+						details = oidcProvider
+					}
+				case model.SSOProviderTypeSAML:
+					var samlProvider model.SAMLProvider
+					if samlProvider, err = bhsaml.GetSAMLProviderBySSOProviderID(s.db, ssoProvider.ID, ctx); err != nil {
+						api.HandleDatabaseError(request, response, err)
+						return
+					} else {
+						details = samlProvider
+					}
+				default:
+					continue // Skip unknown types
+				}
+
+				provider := AuthProvider{
+					ID:      ssoProvider.ID,
+					Name:    ssoProvider.Name,
+					Type:    string(ssoProvider.Type),
+					Slug:    ssoProvider.Slug,
+					Details: details,
+				}
+				providers = append(providers, provider)
+			}
```

Comment:
Couple drive by thoughts: this is making a single db hit per sso provider:
Possible option:
Since both tables are fk'd you might be able to do a more complex sql query that populates it on the fly using table joins. 

Another option: 
Snag all providers from both tables and store em in a  map[sso_id]provider, grab all sso providers then loop in the code instead and compile the response